saq/queue/postgres.py

```diff
@@ -0,0 +1,608 @@
+        }
+
+    async def stats(self, ttl: int = 60) -> QueueStats:
+        stats = self._get_stats()
+        async with self.pool.connection() as conn:
+            await conn.execute(
+                SQL(
+                    """
+                INSERT INTO {stats_table} (worker_id, stats, ttl)
+                VALUES (%(worker_id)s, %(stats)s, %(ttl)s)
+                ON CONFLICT (worker_id) DO UPDATE
+                SET stats = %(stats)s, ttl = %(ttl)s
+                """
+                ).format(stats_table=self.stats_table),
+                {
+                    "worker_id": self.uuid,
+                    "stats": self._dump(stats),
+                    "ttl": seconds(now()) + ttl,
+                },
+            )
+        return stats
+
+    async def count(self, kind: CountKind) -> int:
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            if kind == "queued":
+                await cursor.execute(
+                    SQL(
+                        """
+                    SELECT count(*) FROM {jobs_table}
+                    WHERE status = 'queued'
+                      AND queue = %(queue)s
+                      AND %(now)s >= scheduled
+                    """
+                    ).format(jobs_table=self.jobs_table),
+                    {"queue": self.name, "now": math.ceil(seconds(now()))},
+                )
+            elif kind == "active":
+                await cursor.execute(
+                    SQL(
+                        """
+                    SELECT count(*) FROM {jobs_table}
+                    WHERE status = 'active'
+                      AND queue = %(queue)s
+                    """
+                    ).format(jobs_table=self.jobs_table),
+                    {"queue": self.name},
+                )
+            elif kind == "incomplete":
+                await cursor.execute(
+                    SQL(
+                        """
+                    SELECT count(*) FROM {jobs_table}
+                    WHERE status IN ('new', 'deferred', 'queued', 'active')
+                      AND queue = %(queue)s
+                    """
+                    ).format(jobs_table=self.jobs_table),
+                    {"queue": self.name},
+                )
+            else:
+                raise ValueError("Can't count unknown type {kind}")
+
+            result = await cursor.fetchone()
+            assert result
+            return result[0]
+
+    async def schedule(self, lock: int = 1) -> t.Any: ...
+
+    async def sweep(self, lock: int = 60, abort: float = 5.0) -> list[t.Any]:
+        """Delete jobs past their TTL from the jobs table"""
+        swept = []
+
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                """
+            SELECT pg_try_advisory_xact_lock(%(classid)s, %(objid)s)
+            """,
+                {"classid": SWEEP_LOCK_KEY[0], "objid": SWEEP_LOCK_KEY[1]},
```

Comment:
why is the sweep lock key 1, 1, how did you decide that?

Reply:
I just picked it arbitrarily. I didn't want it to potentially collide with the job lock keys.

Reply:
it doesn't seem like you're using the lock timer here, we only want to sweep every 60 seconds no matter how many workers there are right

Reply:
Ah, you're right. Need to update this.